pkg/controller/knativeserving/openshift/openshift.go

```diff
@@ -40,25 +47,42 @@ const (
 
 	// The secret in which the tls certificate for the autoscaler will be written.
 	autoscalerTlsSecretName = "autoscaler-adapter-tls"
+	// knativeServingInstalledNamespace is the ns where knative serving operator has been installed
+	knativeServingInstalledNamespace = "NAMESPACE"
+	// service monitor created successfully when monitoringLabel added to namespace
+	monitoringLabel = "openshift.io/cluster-monitoring"
 )
 
 var (
+	knativeVersion = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "knative_serving_version",
+		Help: "Installed knative serving version info",
+	}, []string{"version", "namespace", "component"})
```

Comment:
This is too high of a cardinality to be useful in Prometheus. The unique combinations of these label values is potentially an unbounded set, and they should be bounded to at most 10 unique combinations as a general rule of thumb. See the CAUTION section of https://prometheus.io/docs/practices/naming/#labels

We need to rethink what labels are important here and what all the possible combinations of those label values can be.

Reply:
As discussed in Slack, I believe we can drop namespace and component (effectively constant anyway) and only worry about the version string. Over time, that might become an issue assuming one accumulates lots of version. However, Prometheus drops old time series after a certain amount of time (15 days by default) so I'm not sure if this is an issue really.

Reply:
So now do we need to keep only version in labels?

Reply:
I'd propose to go forward with keeping namespace and version, dropping component. Component is effectively already encoded in the name "knative_serving_xxx", namespace is effectively constant currently but might be more fluid in the future, so that prevents stuff from breaking and version is what we need here.